app/src/behaviors/behavior_hold_tap.c

```diff
@@ -280,12 +281,85 @@ static inline char *flavor_str(enum flavor flavor) {
         return "balanced";
     case ZMK_BHV_HOLD_TAP_FLAVOR_TAP_PREFERRED:
         return "tap-preferred";
+    default:
+        return "UNKNOWN FLAVOR";
     }
-    return "UNKNOWN FLAVOR";
 }
 
-static void decide_hold_tap(struct active_hold_tap *hold_tap, enum decision_moment event_type) {
-    if (hold_tap->is_decided) {
+static inline char *status_str(enum status status) {
+    switch (status) {
+    case STATUS_UNDECIDED:
+        return "undecided";
+    case STATUS_HOLD_TIMER:
+        return "hold-timer";
+    case STATUS_HOLD_INTERRUPT:
+        return "hold-interrupt";
+    case STATUS_TAP:
+        return "tap";
+    default:
+        return "UNKNOWN STATUS";
+    }
+}
+
+static inline char *decision_moment_str(enum decision_moment decision_moment) {
+    switch (decision_moment) {
+    case HT_KEY_UP:
+        return "key-up";
+    case HT_OTHER_KEY_DOWN:
+        return "other-key-down";
+    case HT_OTHER_KEY_UP:
+        return "other-key-up";
+    case HT_QUICK_TAP:
+        return "quick-tap";
+    case HT_TIMER_EVENT:
+        return "timer";
+    default:
+        return "UNKNOWN STATUS";
+    }
+}
+
+static int press_binding(struct active_hold_tap *hold_tap) {
+    struct zmk_behavior_binding_event event = {
+        .position = hold_tap->position,
+        .timestamp = hold_tap->timestamp,
+    };
+
+    struct zmk_behavior_binding binding;
+    if (hold_tap->status & STATUS_HOLD) {
+        binding.behavior_dev = hold_tap->config->hold_behavior_dev;
+        binding.param1 = hold_tap->param_hold;
+        binding.param2 = 0;
```

Comment:
The assignments of `0` here shouldn't be necessary, that should be the default value in the struct.

Reply:
As it is currently declared, `binding` is not initialized. It has whatever value happens to be on the stack there. Either this `= 0` needs to stay, or you should change line 327 to something like `binding = {0};` (My preference would be for the latter, because then any new members you add to the struct get initialized too without changing the code.)